hudi-common/src/main/java/org/apache/hudi/expression/Predicates.java

```diff
@@ -0,0 +1,400 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.expression;
+
+import org.apache.hudi.internal.schema.Type;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class Predicates {
+
+  public static True alwaysTrue() {
+    return True.get();
+  }
+
+  public static False alwaysFalse() {
+    return False.get();
+  }
+
+  public static And and(Expression left, Expression right) {
+    return new And(left, right);
+  }
+
+  public static Or or(Expression left, Expression right) {
+    return new Or(left, right);
+  }
+
+  public static BinaryComparison gt(Expression left, Expression right) {
+    return new BinaryComparison(left, Expression.Operator.GT, right);
+  }
+
+  public static BinaryComparison lt(Expression left, Expression right) {
+    return new BinaryComparison(left, Expression.Operator.LT, right);
+  }
+
+  public static BinaryComparison eq(Expression left, Expression right) {
+    return new BinaryComparison(left, Expression.Operator.EQ, right);
+  }
+
+  public static BinaryComparison gteq(Expression left, Expression right) {
+    return new BinaryComparison(left, Expression.Operator.GT_EQ, right);
+  }
+
+  public static BinaryComparison lteq(Expression left, Expression right) {
+    return new BinaryComparison(left, Expression.Operator.LT_EQ, right);
+  }
+
+  public static BinaryComparison startsWith(Expression left, Expression right) {
+    return new BinaryComparison(left, Expression.Operator.STARTS_WITH, right);
+  }
+
+  public static StringContains contains(Expression left, Expression right) {
+    return new StringContains(left, right);
+  }
+
+  public static In in(Expression left, List<Expression> validExpressions) {
+    return new In(left, validExpressions);
+  }
+
+  public static IsNull isNull(Expression child) {
+    return new IsNull(child);
+  }
+
+  public static IsNotNull isNotNull(Expression child) {
+    return new IsNotNull(child);
+  }
+
+  public static Not not(Expression expr) {
+    return new Not(expr);
+  }
+
+  public static class True extends LeafExpression implements Predicate {
```

Comment:
Rename True to TrueExpression as it is very similar to Boolean type. Same for False